### PR TITLE
CMake changes - use openenclave targets and initial install structure

### PR DESCRIPTION
**Pull request by eddyashton**
This adds an install target so that `ninja install` can be used. Currently we install some utilities but not the required headers, so it is not yet possible for an app to build from this install.

Also this replaces our manual paths to openenclave features, based on `OE_PREFIX`, with the targets exported by `find_package(OpenEnclave)`. When we have multiple installs, you can select which is used by tweaking the config passed to `find_package`.

### CODE REVIEW

**Comment by cimetrics** [comment]
cmake_install_target@[3912 aka 20200121.4](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3912&view=results) vs master ewma over 20 builds from [3750](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3750&view=results) to [3908](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3908&view=results)
images

File: cmake/ccf.cmake [hunk 1]
```diff
@@ -0,0 +1,181 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+if((NOT "sgx" IN_LIST TARGET) AND (NOT "virtual" IN_LIST TARGET))
+  message(FATAL_ERROR "Variable list 'TARGET' must be defined and include a valid target")
```
Comment (jumaffre) [inline cmake/ccf.cmake]: Perhaps give an example of a valid target in the error message?

File: cmake/common.cmake [hunk 2]
```diff
@@ -605,10 +412,21 @@ set(CCF_NETWORK_TEST_ARGS
 )
 
 # SNIPPET: Lua generic application
-add_enclave_lib(luagenericenc ${CCF_DIR}/src/apps/luageneric/oe_sign.conf ${CCF_DIR}/src/apps/sample_key.pem SRCS ${CCF_DIR}/src/apps/luageneric/luageneric.cpp)
-
-add_enclave_lib(jsgenericenc ${CCF_DIR}/src/apps/jsgeneric/oe_sign.conf ${CCF_DIR}/src/apps/sample_key.pem SRCS ${CCF_DIR}/src/apps/jsgeneric/jsgeneric.cpp)
+add_enclave_lib(luagenericenc
```
Comment (jumaffre) [inline cmake/common.cmake]: This will need the multi-line snippet comment to be displayed properly in the docs.

File: cmake/ccf.cmake [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      ${EVERCRYPT_INC}
+      ${CMAKE_CURRENT_BINARY_DIR}
+      ${QUICKJS_INC}
+      ${OE_INCLUDEDIR} # Virtual libraries don't link against OE, but do share includes
+    )
+    add_dependencies(${virt_name} flatbuffers)
+
+    if (PBFT)
+      target_link_libraries(${virt_name} PRIVATE
+        -Wl,--allow-multiple-definition #TODO(#important): This is unfortunate
```
Comment (olgavrou) [inline cmake/ccf.cmake]: Not really part of this PR but since we removed the `libbyz/keypair.h` I think this doesn't have to be unfortunate anymore (I think we can remove the `-Wl, --alow-multiple-definitions`)

File: cmake/ccf.cmake (hunk 1 above)
Comment (eddyashton) [reply cmake/ccf.cmake]: Sure, done!

File: cmake/common.cmake (hunk 2 above)
Comment (eddyashton) [reply cmake/common.cmake]: Good catch - have updated the docs as well.

File: cmake/ccf.cmake (hunk 3 above)
Comment (eddyashton) [reply cmake/ccf.cmake]: Nice - have removed this unpleasantness entirely.

File: CMakeLists.txt [hunk 4]
```diff
@@ -11,6 +11,31 @@ set(TESTS_SUFFIX $ENV{TESTS_SUFFIX})
 message(STATUS "Setting TESTS_SUFFIX on performance tests to '${TESTS_SUFFIX}'")
 set(ENV{BETTER_EXCEPTIONS} 1)
 
+find_package(Git)
+if(GIT_FOUND)
```
Comment (achamayou) [inline CMakeLists.txt]: That's not great for someone checking out the tarballs, which is by far the most likely scenario for actual releases.

Comment (eddyashton) [reply CMakeLists.txt]: That's a good point. I've removed this, and we now just retrieve the version from the `PROJECT_VERSION` CMake variable, set [with this](https://cmake.org/cmake/help/latest/command/project.html#options).

**Comment by achamayou** [comment]
This closed #52 as well.
